integration/test/Test/MLS/Unreachable.hs

```diff
@@ -0,0 +1,98 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2023 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Test.MLS.Unreachable where
+
+import API.Galley
+import Control.Monad.Codensity
+import Control.Monad.Reader
+import MLS.Util
+import Notifications
+import SetupHelpers
+import Testlib.Prelude
+import Testlib.ResourcePool
+
+testAddUsersSomeReachable :: HasCallStack => App ()
+testAddUsersSomeReachable = do
+  (addCommit, d) <- startDynamicBackends [mempty] $ \[thirdDomain] -> do
+    ownDomain <- make OwnDomain & asString
+    otherDomain <- make OtherDomain & asString
+    [alice, bob, charlie] <- createAndConnectUsers [ownDomain, otherDomain, thirdDomain]
+
+    [alice1, bob1, charlie1] <- traverse (createMLSClient def) [alice, bob, charlie]
+    traverse_ uploadNewKeyPackage [bob1, charlie1]
+    void $ createNewGroup alice1
+    void $ withWebSocket bob $ \ws -> do
+      void $ createAddCommit alice1 [bob] >>= sendAndConsumeCommitBundle
+      awaitMatch 10 isMemberJoinNotif ws
+    mp <- createAddCommit alice1 [charlie]
+    pure (mp, thirdDomain)
+
+  -- try adding Charlie now that his backend is unreachable
+  bindResponse (postMLSCommitBundle addCommit.sender (mkBundle addCommit)) $ \resp -> do
+    resp.status `shouldMatchInt` 533
+    (resp.json %. "unreachable_backends" & asList) `shouldMatch` [d]
```

Comment:
why not `shouldMatchSet` as below?

Reply:
For no particular reason.

I believe this way it is fine because it's only one list element so there's no concern about what element comes where.